### DISCUSSION TITLE
How to set specific compiler for vcpkg.exe?

### DISCUSSION THREAD

**Discussion by felixf4xu** [microsoft/vcpkg#40248] [Q&A] [answered]
Hi,

When I run `vcpkg.exe install package`, how to choose the compiler? I have `vc++` and `gcc` installed.

**Comment by dg0yt** [comment, marked as answer]
Choose triplets. x64-window... is for MSVC, x64-mingw... is for mingw gcc.
Start with setting the host triplet when you want to avoid a mix of MSVC and gcc due to host dependencies.
https://learn.microsoft.com/vcpkg/concepts/triplets

**Reply by felixf4xu** [reply]
oh, too bad
>python3 is only supported on '!uwp & !mingw'

python3 is needed for many packages, is it's not available on `mingw`, many other packages can't be build.

But clearly I remember that on msys2/mingw, python3 is available.

**Reply by dg0yt** [reply]
MSYS2: This is the list of the 149 patches they have to apply for python3:
https://github.com/msys2/MINGW-packages/tree/master/mingw-w64-python
It would be easier if upstream would support it.
